tools/SetupFlow/DevHome.SetupFlow/Models/RepositoryProvider.cs

```diff
@@ -261,27 +267,41 @@ public IEnumerable<IDeveloperId> GetAllLoggedInAccounts()
         return developerIdsResult.DeveloperIds;
     }
 
-    /// <summary>
-    /// Gets all the repositories an account has for this provider.
-    /// </summary>
-    /// <param name="developerId">The account to search in.</param>
-    /// <param name="searchInputs">The key/value pairs that the extension will use to search for repos.</param>
-    /// <returns>A collection of repositories.  May be empty</returns>
-    public IEnumerable<IRepository> GetAllRepositories(IDeveloperId developerId, Dictionary<string, string> searchInputs)
+    public RepositorySearchInformation SearchForRepositories(IDeveloperId developerId, Dictionary<string, string> searchInputs)
     {
-        TelemetryFactory.Get<ITelemetry>().Log("RepoTool_GetAllRepos_Event", LogLevel.Critical, new GetReposEvent("CallingExtension", _repositoryProvider.DisplayName, developerId));
+        TelemetryFactory.Get<ITelemetry>().Log("RepoTool_SearchForRepos_Event", LogLevel.Critical, new GetReposEvent("CallingExtension", _repositoryProvider.DisplayName, developerId));
 
-        RepositoriesResult result;
+        var repoSearchInformation = new RepositorySearchInformation();
         try
         {
-            if (IsSearchingEnabled())
+            if (IsSearchingEnabled() && searchInputs != null)
             {
                 var repositoryProvider2 = _repositoryProvider as IRepositoryProvider2;

```

Comment:
If _repositoryProvider does not implement IRepositoryProvider2 the next line will throw an exception. Is that by design?
I'd think that older/existing providers that don't implement IRepositoryProvider2 should continue working without throwing/reporting errors (but with limited functionality).

Reply:
`IsSearchingEnabled` also checks if `_repositoryProvider` implements `IRepositoryProvider2`  However, I can add another check here as well.